docs/input.md

```diff
@@ -129,29 +135,118 @@ description: |
 
-### `min`
+#### `min`
 
 Minimum value for constraining the value range. The `min` field is only valid for `type=integer` and `type=float`. Setting a minimum value is optional and can be omitted.  
 Note that if a `max` value is additionally specified for the parameter, `min` must be lower than `max`.
 
-### `max`
+#### `max`
 
 Maximum value for constraining the value range. The `max` field is only valid for `type=integer` and `type=float`. Setting a maximum value is optional and can be omitted.  
 Note that if a `min` value is additionally specified for the parameter, `max` must be higher than `min`.
 
-### `optional`
+#### `optional`
 
 Boolean field which defaults to `false`. If set to `optional=true`, the parameter is not required by the tool. This implies, that the tool implementation can handle a `parameters.json` in which the `Parameter` is entirely missing.
 
-### `default`
+#### `default`
 
-The `default` field is of same data type as the `Parameter` itself. If a default value is set, the tool-framework is required to inject this parameter into the `parameters.json`, as the tool will treat the default like any other non-optional parameter.  
+The `default` field is of the same data type as the `Parameter` itself. If a default value is set, the tool-framework is required to inject this parameter into the `parameters.json`, as the tool will treat the default like any other non-optional parameter.  
 Note, that default parameters are only parsed if they are not set as `optional=true`.
 
+
+## Data: File specification
+
+All input `Data` is described in a data block in the `/src/tool.yml` file.
+All sets of input data are collected as the mandatory `tools.<tool_name>.data` block:
+
+```yaml
+tools:
+  foobar:
+    parameters:
+      ...
+    data:
+      foo_data:
+        ...
+```
+
+Refer to the section below to learn about mandatory and optional fields for `Data`.
+
+
+### Fields
+
+The following section defines all mandatory and optional fields of a `Data` entity.
+
+#### `path`
+
+The `path` field is the only mandatory field. Each input data needs a path, as input data
+is always given to the tool in a file- or folder-based way.
+Just as for paramters of `type=file`, the library used for parsing parameterizations and 
+data loads the file into memory and passes the respective datastructure to the tool, which 
+makes the tool independent of specific data files. 
+If the file format / extension is not supported by the parsing library, file paths are 
+passed just as strings, the parsing library will not attempt to load the file. 
+
+There are a number of file types, which are loaded by default:
+
+| file extension | Python |  R  |  Matlab |  NodeJS  |
+| ---------------|--------|-----|---------|----------| 
+| .dat  |  `numpy.array` | `vector` | `matrix`  | `number[][]` | 
+| .csv  |  `pandas.DataFrame` | `data.frame` |  `matrix` |  `number[][]` |
+
+#### `format`
```

Comment:
hmm, format might be mis-leading here. What do you think about `mime` or `extension`?

Reply:
I like `extension`.